machineconfiguration/v1/types.go

```diff
@@ -398,6 +398,24 @@ type MachineConfigPoolSpec struct {
 	// The targeted MachineConfig object for the machine config pool.
 	// +optional
 	Configuration MachineConfigPoolStatusConfiguration `json:"configuration"`
+
+	// pinnedImageSets is a list of PinnedImageSetRef objects that should be
+	// applied to the nodes in this pool.
+	// +openshift:enable:FeatureGate=PinnedImages
+	// +optional
+	// +listType=atomic
+	PinnedImageSets []PinnedImageSetRef `json:"pinnedImageSets"`
```

Comment:
One naive question: `PinnedImageSetRef` contains one element with type string. Is there a reason to make PinnedImageSetRef a struct instead of directly defining as string?

Reply:
It allows for potential future expansion. For example, perhaps in the future we want to `errorIfMissing` or set a priority or something else for each referenced image set